packages/web3-core-method/src/observers/TransactionObserver.js

```diff
@@ -158,8 +159,15 @@ export default class TransactionObserver {
                             this.emitNext(receipt, observer);
                         }
                     } else {
-                        this.lastBlock = await this.getBlockByNumber(receipt.blockNumber);
-                        this.confirmations++;
+                        // on parity nodes you can get the receipt without it being mined
+                        // so the receipt may not have a block number at this point.
+                        // we should check that the blockNumber is defined before we do this call
+                        // geth nodes only return the receipt once mined as the spec states.
+                        if (receipt.blockNumber) {
```

Comment:
the ``receipt.blockNumber`` check can be moved on line 152 and this big inline comment can be changed to: ``// blockNumber check for parity``.

Reply:
yeah good shout we should only emit the receipt once mined i think else Dapps could get super confused. Will change

Reply:
Ok Updated i have made the comment smaller but i have still explained the point. Think its critical to explain why as people will forget.